### DISCUSSION TITLE
Jarvis Map

### DISCUSSION THREAD

**Discussion by kilian-f-za** [Zefau/ioBroker.jarvis#1785] [Q&A]
Ich habe im Jarvis das Map Widget hinzugefügt, soweit funktioniert alles aber die Datenpunkte werden nicht aktualisiert. Wenn sich der Standort ändert muss ich jedesmal die Seite komplett neu laden. Gibt es eine Möglichkeit einen auto refresh oder so einzufügen?

**Comment by mcuiobroker** [comment]
Warum 2x eingestellt? Welche Version?

**Reply by kilian-f-za** [reply]
Die aktuelle Version aus dem iobroker 3.0.11

**Comment by mcuiobroker** [comment]
https://mcuiobroker.gitbook.io/jarvis-infos/jarvis/system-effektprog./v3-systemprogramme/setdefaultposition-map-modul

**Reply by kilian-f-za** [reply]
Vielen Dank, genau das habe ich gesucht. Ist geplant die Funktion noch zu erweitern für mehrere Karten?

**Reply by kilian-f-za** [reply]
Ich habe festgestellt, dass bei mehren Follow Positionen die Karte zwar zu der neuen Position springt aber die Position (also die Stecknadel von der z.B. PosA) nicht anzeigt. Muss ich da meinen Browser Cache oder so leeren?

**Reply by mcuiobroker** [reply]
Nein.
Man kann theoretisch das Script kopieren und die Hauptzeile anpassen:
`let posDefaultDP = '0_userdata.0.jarvis.positions.posDefault';` auf einen anderen Wert posDefault2 oder so.
Das Problem ist nur, wenn sich gleichzeitig die Werte ändern, dann werden von 2 Scripten gleichzeitig die devices geändert und das kann schon auch daneben gehen. Deswegen immer die DPs sichern!!!!
Oder meinst du die 4 Positionen geleichzeitig auf mehren Karten anzuzeigen?
Dann keine Karte auswählen. Dann kommt zwar die beschriebene Fehlermeldung, aber es müsste auf allen Karten angezeigt werden.

**Reply by mcuiobroker** [reply]
Zeig mal bitte die Objekt Einstellungen

**Reply by kilian-f-za** [reply]
Okay, danke. 

Ich meinte 4 andere Positionen auf einer anderen Karte anzeigen. Das jetzt ja wie du es beschrieben hast mit der Änderung in der Hauptzeile.

**Reply by kilian-f-za** [reply]
> Zeig mal bitte die Objekt Einstellungen

Im iobroker oder in der config von jarvis?

**Reply by mcuiobroker** [reply]
iobroker-objekte

0_userdata.0.jarvis.positions

**Reply by mcuiobroker** [reply]
>Muss ich da meinen Browser Cache oder so leeren?

Nein. Man muss darauf achten ,dass man ca. 5 Sekunden Abstand zwischen den Änderungen hat sonst zieht es nicht, teilweise. War bei mir so.

**Reply by mcuiobroker** [reply]
> die Position (also die Stecknadel von der z.B. PosA) nicht

Also bei mir springt auch die Stecknadel um. Man sollte den Zoom so einstellen, dass man sie immer sehen kann.

**Reply by kilian-f-za** [reply]
![image](https://user-images.githubusercontent.com/87258544/180603635-d596cca4-b6f1-4960-a47a-868852794ded.png)

Meinst du die?
Das kann sein, manchmal geht es aber wenn der standort meiner Gerät aktualisiert wird manchma halt nicht. Dann muss ich da einen sleep oder so einbauen. Meine Standort Daten werden zu festen Zeiten synchronisiert und deshalb kommen die fast zeitgleich

**Reply by kilian-f-za** [reply]
> Also bei mir springt auch die Stecknadel um. Man sollte den Zoom so einstellen, dass man sie immer sehen kann.

Okay, das manchmal schwierig. Die Positionen sind manchmal weit auseinander

**Reply by mcuiobroker** [reply]
Diese:
![image](https://user-images.githubusercontent.com/84818934/180603808-d5971cc0-fbc9-4ad1-a659-760a730085a7.png)

**Reply by kilian-f-za** [reply]
<img width="1538" alt="image" src="https://user-images.githubusercontent.com/87258544/180603874-9f41e1ce-49f1-4940-befd-b019b84050e7.png">

**Reply by mcuiobroker** [reply]
Die Einstellungen sind korrekt.

**Reply by kilian-f-za** [reply]
Ok, ich werde es erstmal beobachten. 

Gibt es die Möglichkeit den Zoom mit einzustellen?

**Reply by mcuiobroker** [reply]
Muss nachher mal schauen. Ich glaube nicht. Bin gerade unterwegs.

**Reply by kilian-f-za** [reply]
Ich habe es jetzt probiert. Wenn die Positionen ziemlich zeitnah aktualisiert werden geht es nicht. Es funktioniert dann auch nicht mehr die Tooltips womit man zu den jeweiligen Personen springen kann.

**Reply by mcuiobroker** [reply]
defaultZoom könnte man auch setzen:
![image](https://user-images.githubusercontent.com/84818934/180618781-3869675c-7bd9-4b03-ba31-bffde94317cf.png)
Wie hast du dir das vorgestellt mit der Zoom-Änderung?

**Reply by kilian-f-za** [reply]
> defaultZoom könnte man auch setzen:
> ![image](https://user-images.githubusercontent.com/84818934/180618781-3869675c-7bd9-4b03-ba31-bffde94317cf.png)
> Wie hast du dir das vorgestellt mit der Zoom-Änderung?

Das der Zoom reduziert wird wenn die FollowPos weiter auseinander liegen und wenn die Pos wieder nah beisammen sind weiter reinzommt. Ist das umsetzbar?

**Reply by kilian-f-za** [reply]
> >Muss ich da meinen Browser Cache oder so leeren?
> 
> Nein. Man muss darauf achten ,dass man ca. 5 Sekunden Abstand zwischen den Änderungen hat sonst zieht es nicht, teilweise. War bei mir so.

Kann man da eine Wartenanweisung einbringen, dass nur alle paar Sekunden die Änderungen übernommen werden?

**Reply by mcuiobroker** [reply]
Habe jetzt eine zusätzliche Option eingebaut. Schedule kann gesetzt werden in [min], reicht das?
https://mcuiobroker.gitbook.io/jarvis-infos/jarvis/system-effektprog./v3-systemprogramme/setdefaultposition-map-modul/javascript-setdefaultposition-v1.0.1-test

**Reply by kilian-f-za** [reply]
Super, ja sieht gut aus. Ich werde es testen. Die Zoomstufen können ziemlich weit oben im Script angepasst werden oder?

**Reply by mcuiobroker** [reply]
Nein. Ist nur ein Versuchsarray. Hat keinen Einfluss.

**Reply by kilian-f-za** [reply]
OK, danke für die schnelle Anpassung des Scripts. Ich werde es testen. Ich glaube der schedule ist gut.

**Comment by Zefau** [comment]
Mit v3.1 ist es nun möglich einem Gerät zu folgen, siehe https://github.com/Zefau/ioBroker.jarvis/issues/1798#issuecomment-1242813216
